Drivers/rc_receiver/Tests/sbus_gtest.cpp

```diff
@@ -0,0 +1,98 @@
+#include <gtest/gtest.h>
+#include <fff.h>
+#include "sbus_receiver.hpp"
+
+DEFINE_FFF_GLOBALS;
+
+/*
+    Faking the HAL function in this unit test to do nothing
+    The recieved data will be manually added to test cases
+*/
+FAKE_VALUE_FUNC(HAL_StatusTypeDef, HAL_UART_Receive_DMA, UART_HandleTypeDef*, uint8_t*, uint16_t );
+
+TEST (SBUS_Receiver_Test, RawDataToSBus){
+    const uint8_t in_bytes[SBUS_FRAME_SIZE] = {
+        HEADER_, 0x80, 0x08, 0x84,
+        0x20, 0x06, 0x41, 0x88, 0x42, 0x18,
+        0xE2, 0x10, 0x88, 0x48, 0x84, 0x22,
+        0x16, 0xC1, 0x88, 0x46, 0x38, 0xE2,
+        0x11, 0x00, FOOTER_
+    };
```

Comment:
I think this is a great start, but I think it would be good to see some more test cases. Perhaps thing about what values are edge cases and ensure that the code still works correctly.